### DISCUSSION TITLE
Can version increase check be done earlier

### DISCUSSION THREAD

**Discussion by vogella** [eclipse-tycho/tycho#1018] [Ideas]
Tycho provides the very useful check to check if a version must be increased.

AFAICS the version increase check is done after the compile and test steps. This leads to longer verification build times. Would it be possible to move this check before the compile and test phase? For example, for https://github.com/eclipse-pde/eclipse.pde/pull/155 I always had to wait ~ 40 min to see if another version needs an increase.

**Comment by laeubi** [comment]
> This leads to longer verification build times. Would it be possible to move this check before the compile and test phase?

How should this be performed if nothing is there to compare? :-)

> I always had to wait ~ 40 min

An sustainable part of the build time is target resolution and test execution but not compilation. You could improve build times by enable maven parallel builds considerable.

**Reply by vogella** [reply]
Thanks @laeubi

**Reply by laeubi** [reply]
FYI:
- https://github.com/eclipse/tycho/issues/1034
